### PR TITLE
add open options

### PR DESCRIPTION
**Pull request by BusyJay**
This PR adds 3 types of open options for building db:
- Open db with the default cf only,
- Open db with multiple cfs,
- Open titan db with multiple cfs.

### CODE REVIEW

File: tirocks/src/db/open_options.rs [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+    fn add_cf_internal(
+        &mut self,
+        name: impl Into<String>,
+        cf: CfOptions,
+        ttl: Option<i32>,
+    ) -> &mut Self {
+        let name = name.into();
+        if let Some(ttl) = ttl {
+            if self.cfs.len() != self.ttl.len() {
+                panic!("ttl should be specified for all cf.");
```
Comment (tabokie) [inline tirocks/src/db/open_options.rs:115]: Why?

File: tirocks/src/db/open_options.rs [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+
+    fn open(&self, path: &Path) -> Result<Db> {
+        let mut comparator = vec![];
+        let mut env = None;
+        let mut handles = Vec::with_capacity(0);
+        let (ptr, is_titan) =
+            unsafe { self.open_raw(&mut comparator, &mut env, &mut handles, path) }?;
+        let handles = if handles.is_empty() {
+            vec![unsafe {
+                let ptr = tirocks_sys::crocksdb_get_default_column_family(ptr);
```
Comment (tabokie) [inline tirocks/src/db/open_options.rs:213]: If I remember correctly, you can't use this handle as if it's created from `crocksdb_open_column_families`. i.e. you can't destroy it.

Comment (BusyJay) [reply tirocks/src/db/open_options.rs:213]: That's the reason why `RefCountedCfHandle` is constructed using `false`, which means tirocks should not destroy the handle.

**Review by tabokie — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**